### PR TITLE
74 job role identifies a social worker approved mental health 152 frontend

### PR DESCRIPTION
**Pull request by adamfaryna**
https://trello.com/c/96pm3aYq/74-job-role-identifies-a-social-worker-approved-mental-health-152

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -192,23 +194,33 @@ const routes: Routes = [
       {
         path: 'create-staff-record/:id',
         component: CreateStaffRecordComponent,
-        canLoad: [AuthGuard],
+        canLoad: [AuthGuard]
       },
       {
         path: 'create-staff-record',
```
Comment (wozzer72) [inline src/app/app-routing.module.ts]: Seeing what you've done now with MentalHealth component and other jobs, an "optional :id" passed to the route, it doesn't make sense now to have two separate routes for `CreateStaffRecordComponent` - and indeed, the name of the component no longer makes sense - in that this one component is both create and edit staff record. A little refactoring now, even though it is not within scope of this Trello card, would make good sense.

File: src/app/core/model/job.model.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-export interface Job {
-  id: number,
-  title: string
+export class Job {
+  id?: number
```
Comment (wozzer72) [inline src/app/core/model/job.model.ts]: Why now are all these optional?

File: src/app/features/workers/create-staff-record/create-staff-record.component.ts [hunk 3]
```diff
@@ -58,30 +57,30 @@ export class CreateStaffRecordComponent implements OnInit, OnDestroy {
 
   async submitHandler() {
     try {
-      await this.saveHandler()
+      const res = await this.saveHandler()
 
       if (this.isSocialWorkerSelected()) {
-        this.router.navigate(["/worker/mental-health"])
+        this.router.navigate([`/worker/mental-health/${res.uid}`])
```
Comment (wozzer72) [inline src/app/features/workers/create-staff-record/create-staff-record.component.ts]: No such destination: `/worker/mental-health/${res.uid}`.

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (adamfaryna) [reply src/app/app-routing.module.ts]: Actually `workerId` parameter is not optional for any routes following CreateStaffRecord, it's part of the route path and if absent the path will not match.

Sure, I can amend the route path for CreateStaffRecord in this PR, but as you said it out of scope of this PR and should been reported in PR related to CreateStaffRecord.

File: src/app/core/model/job.model.ts (hunk 2 above)
Comment (adamfaryna) [reply src/app/core/model/job.model.ts]: ok

File: src/assets/scss/forms.scss [hunk 4]
```diff
@@ -116,7 +116,7 @@ form,
   clear: left;
   position: relative;
   padding: 8px 0 0 45px;
-  margin-bottom: 10px;
+  margin-bottom: 20px;
```
[the remark this answers is not in the document]
Comment (adamfaryna) [reply src/assets/scss/forms.scss]: The reason is: there is no enough space between radio buttons, and to stay consistent with the designs we have to amend it.

**Comment by adamfaryna** [comment]
> I'm likely the 'blank' templates on other jobs and main job start date; it makes the focus of this review easier.
> 
> I'm happy to approve and merge this change. But I have just two comments which I would appreciate your feedback on.
> 
> The first about re-factoring `create-staff-record` component I am happy to record against the Trello card itself, but visually, the comment make sense here. So would value your feedback on that so I can add a comment to the other trello card.
> 
> And the other comment regarding the Job interface is minor; I'm happy to accept no change.

The comments were addressed, can you approve this PR now?

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (wozzer72) [reply src/app/app-routing.module.ts]: So please raised a trello card to get CreateStaffRecord refactored.

**Comment by wozzer72** [comment]
I won't approve this because:
1. My comment `src/assets/scss/forms.scss` was resolved - without giving me the chance to review the feedback on the comment I raised. 
There are wider implications that just this Trello card in the change to `src/assets/scss/forms.scss`.

The author needs to understand and seek forward agreement on any change that has a wider impact, and especially when choosing as an argument that other work is not in scope, should seriously consider if the change done here is within scope.

And never, resolve reviewer's comments; you must aswell approve and merge your own changes.

2. The refactoring of "create-staff-worker" needs to be captured as technical debt in a Trello card.

Until then. Unapproved.

**Comment by adamfaryna** [comment]
> I won't approve this because:
> 
> 1. My comment `src/assets/scss/forms.scss` was resolved - without giving me the chance to review the feedback on the comment I raised.
>    There are wider implications that just this Trello card in the change to `src/assets/scss/forms.scss`.
> 
> The author needs to understand and seek forward agreement on any change that has a wider impact, and especially when choosing as an argument that other work is not in scope, should seriously consider if the change done here is within scope.
> 
> And never, resolve reviewer's comments; you must aswell approve and merge your own changes.
> 
> 1. The refactoring of "create-staff-worker" needs to be captured as technical debt in a Trello card.
> 
> Until then. Unapproved.

1. @wozzer72 you can review it now. I "unresolved" the conversarsation about this code block, so you can put your concerns there.

2. @wozzer72 there were no "refactoring of create-staff-worker" needs, there is no mention about create-staff-worker trello card how we handle URI paths. If it blocks you from approving this PR, I can remove related code block, because I added it only for your request to this PR.

**Comment by adamfaryna** [comment]
[Job Role identifies a Social Worker - Approved Mental Health #152](https://trello.com/c/96pm3aYq/37-job-role-identifies-a-social-worker-approved-mental-health-152)

**Comment by adamfaryna** [comment]
[Job Role identifies a Social Worker - Approved Mental Health #152](https://trello.com/c/96pm3aYq/37-job-role-identifies-a-social-worker-approved-mental-health-152)

**Pull request merged**